lib/ansible/module_utils/network/netscaler/netscaler.py

```diff
@@ -205,25 +210,56 @@ def ensure_feature_is_enabled(client, feature_str):
 def get_nitro_client(module):
     from nssrc.com.citrix.netscaler.nitro.service.nitro_service import nitro_service
 
-    client = nitro_service(module.params['nsip'], module.params['nitro_protocol'])
-    client.set_credential(module.params['nitro_user'], module.params['nitro_pass'])
+    if module.params['mas_proxy_call']:
+
+        try:
+            from nssrc.com.citrix.netscaler.nitro.service.MasContext import MasContext
+        except ImportError as e:
+            module.fail_json(msg='The currently installed nitro python SDK does not support MAS proxied calls')
+
+        if module.params['mas_auth_token'] is None:
+            module.fail_json(msg='You must provide a mas authentication token to proxy a call trough MAS')
+        if module.params['instance_ip'] is None:
+            module.fail_json(msg='You must provide  the target NS instance ip to proxy a call trough MAS')
+
+        masCtx = MasContext(
+            module.params['mas_ip'],
+            module.params['mas_auth_token'],
+            module.params['instance_ip'],
+        )
+
+        client = nitro_service(masCtx, module.params['nitro_protocol'])
+    else:
+        if module.params['nitro_user'] is None:
+            module.fail_json(msg='You must provide a valid nitro user name')
+
+        if module.params['nitro_pass'] is None:
+            module.fail_json(msg='You must provide a valid nitro password')
+
+        client = nitro_service(module.params['nsip'], module.params['nitro_protocol'])
+        client.set_credential(module.params['nitro_user'], module.params['nitro_pass'])
+
     client.timeout = float(module.params['nitro_timeout'])
     client.certvalidation = module.params['validate_certs']
+
     return client
 
 
 netscaler_common_arguments = dict(
```

Comment:
Might be worth adding something like the following (not tested)

```
required_one_of=[['nitro_pass', 'nitro_auth_token']],
mutually_exclusive=(['nitro_pass', 'nitro_auth_token'],),
required_together=(['nitro_user', 'nitro_pass'],),
```
See https://github.com/gundalow/ansible/blob/docs-argspec/docs/docsite/rst/dev_guide/module_utils/basic.rst

Reply:
Since the AnsibleModule instantiation happens on each module I have changed the one module included in this PR `netscaler_save_config`.

I will be adding these arguments in subsequent modules as well.